File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.swift
class Solution {
    func maximumUnits(_ boxTypes: [[Int]], _ truckSize: Int) -> Int {
        let sorted = boxTypes.sorted(by: {
            $0[1] > $1[1]
        })
        var left = truckSize
        var sum = 0
        
        sorted.forEach{
            var box = $0[0], unit = $0[1]
            while left > 0 {
                left -= 1
                box -= 1
                sum += unit

                if box == 0 {
                    break
                }
            }
        }

        return sum  
    }
}

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a, b) -> b[1] - a[1]);
        int sum = 0;
        int left = truckSize;


        for (int[] boxType : boxTypes) {
            int box = boxType[0];
            int unit = boxType[1];
            while (left > 0) {
                box -= 1;
                left -= 1;
                sum += unit;

                if (box == 0) {
                    break;
                }
            }
            
        }
        
        return sum;
    }
}